### PR TITLE
fix(listitem): fix row actions issue

### PR DESCRIPTION
**Pull request by scottdickerson**
Closes #

**Summary**

- Fixes two issues,:
one the row actions cause the list values to overlap and the padding wasn't giving enough space
two fixes the warning thrown by the DataSeriesFormItem

**Acceptance Test (how to verify the PR)**

- Check the updated story to see the longer list item and notice the proptypes error goes away on the new data series form

### CODE REVIEW

File: src/components/List/ListItem/ListItem.jsx [hunk 1]
```diff
@@ -167,9 +167,11 @@ const ListItem = ({
       </div>
     ) : null;
 
+  const hasRowActions =
```
Comment (bjornalm) [inline src/components/List/ListItem/ListItem.jsx:170]: nitty gritty, but can't this just be `typeof rowActions === 'function' || rowActions?.length`?

File: src/components/List/ListItem/_list-item.scss [hunk 2]
```diff
@@ -175,7 +175,7 @@
         text-overflow: ellipsis;
         padding-right: $spacing-09;
         &__with-actions {
-          padding-right: 0;
+          padding-right: $spacing-06; // save room for the action button
```
Comment (bjornalm) [inline src/components/List/ListItem/_list-item.scss:178]: You need even more room. The `bx--overflow-menu` is 2rem so you'll need at least `$spacing-07`

**Review by bjornalm — CHANGES_REQUESTED** [review]

File: src/components/List/ListItem/_list-item.scss (hunk 2 above)
Comment (scottdickerson) [reply src/components/List/ListItem/_list-item.scss:178]: for some reason that seems to work for me, I think the ellipses padding helps

**Review by bjornalm — APPROVED** [review]
Nice job!

**Review by tay1orjones — APPROVED** [review]
looks great!

**Pull request merged**